packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -76,6 +85,22 @@ export const displayTxResult = ({
       );
     }
 
+    if (displayContent instanceof CairoOption) {
+      if (displayContent.isNone()) {
+        JSON.stringify({ None: displayContent.unwrap() }, replacer);
+      }
+
+      return JSON.stringify({ Some: displayContent.unwrap() }, replacer);
+    }
+
+    if (displayContent instanceof CairoResult) {
+      if (displayContent.isOk()) {
+        JSON.stringify({ Ok: displayContent.unwrap() }, replacer);
+      }
+
+      return JSON.stringify({ Err: displayContent.unwrap() }, replacer);
+    }
+
```

Comment:
this part will have code conflictions with latest decoding logic, would you mind to do a code rebase and also add some unit tests to cover this part as well.

Reply:
got it, i was rushing when doing this PR so that some guys in the HH wanted to try it. But now that it's more lax i can start adding UTs as well, will fix!

Reply:
i just used your changes and seems to be working good, will add the UTs